lib/spack/spack/binary_distribution.py

```diff
@@ -676,6 +676,8 @@ def get_specs(force=False, use_arch=False, names=[]):
     if use_arch:
         arch_pattern = '(%s-%s-[^-]*)' % (arch.platform, arch.os)
 
+    if not names:
```

Comment:
```suggestion
    if names is not None:
```
https://www.python.org/dev/peps/pep-0008/#programming-recommendations:

> Comparisons to singletons like None should always be done with is or is not, never the equality operators.
>
> Also, beware of writing if x when you really mean if x is not None -- e.g. when testing whether a variable or argument that defaults to None was set to some other value. The other value might have a type (such as a container) that could be false in a boolean context!